lib/internal/http.js

```diff
@@ -31,9 +33,33 @@ function ondrain() {
   if (this._httpMessage) this._httpMessage.emit('drain');
 }
 
+function now() {
+  const hr = process.hrtime();
+  return hr[0] * 1000 + hr[1] / 1e6;
+}
+
+class HttpRequestTiming {
+  constructor(statistics) {
+    this.name = 'HttpRequest';
+    this.entryType = 'http';
+    this.duration = statistics.end_time - statistics.start_time;
+    this.startTime = statistics.start_time;
+  }
+}
+
+Object.setPrototypeOf(
+  HttpRequestTiming.prototype, PerformanceEntry.prototype);
+Object.setPrototypeOf(HttpRequestTiming, PerformanceEntry);
```

Comment:
Is there a reason not to use `class HttpRequestTiming extends PerformanceEntry` directly?

Reply:
I would agree with you but the [`PerformanceNodeTiming` code](https://github.com/nodejs/node/blob/master/lib/perf_hooks.js#L216) is implemented like this, so i would tend to use the same behavior here.

Reply:
@vmarchaud Let’s fix that :) https://github.com/nodejs/node/pull/28495